### ISSUE TITLE
Two instances of johnny not synchronized

### ISSUE THREAD

**Issue by math07** [shinnok/johnny#7]
Let's use github issues to keep track of things that are easy to loose on the list.
This TODO might need more than a few lines so it might need a separate task if it is judged important. @AlekseyCherepanov Did you qualify that as something important ?
From MainWindow::on_actionStart_Attack_triggered()
  // TODO: Saving so two instances of johnny overwrite description
    //       but not .rec so they become not synchronized.
    QFile description(m_session + ".johnny");
    if (!description.open(QIODevice::WriteOnly | QIODevice::Text)) {
        QMessageBox::critical(
            this,
            tr("Johnny"),
            tr("Johnny could not open file to save session description!"));
        return;
    }

**Comment by AlekseyCherepanov** [comment]
On Wed, May 13, 2015 at 12:20:18PM -0700, math07 wrote:
> This TODO will need more than a few lines so it might need a separate task if it is judged important. @AlekseyCherepanov Did you qualify that as something important ?
>   // TODO: Saving so two instances of johnny overwrite description
>     //       but not .rec so they become not synchronized.
>     QFile description(m_session + ".johnny");
>     if (!description.open(QIODevice::WriteOnly | QIODevice::Text)) {
>         QMessageBox::critical(
>             this,
>             tr("Johnny"),
>             tr("Johnny could not open file to save session description!"));
>         return;
>     }

Currently running of 2 instances of Johnny is not supported. Later it
may be prohibited at all to run all sessions in one instance.

The todo is about .johnny and .rec about different files. The code
looks like 2 johnnies should start attack almost at the same time: #1
creates .johnny but .rec is not created yet, in parallel #2 overwrites
.johnny but does not remove .rec and #2's john fails so the .rec file
is not overwritten, so we have .johnny from #2 and .rec from #1.
That's the race. It does not seem possible with regular user (though
some scripts that automate gui may be hurt, unrealistic though). It
does not seem harmful if user intentionally overwrites .johnny with
wrong content.

So it should not be important, but it relates to a separate feature. I
guess it is "multiple cracking sessions management" in 1.6.1. Shinnok
may want to comment on this.

Thanks!

-- 
Regards,
Aleksey Cherepanov

**Comment by shinnok** [comment]
I'm not sure I understand what this task is supposed to be. Is it regarding running multiple instances of Johnny? If so I don't think we really need that if we implement cracking sessions.

**Comment by math07** [comment]
Yes, it's regarding two instances of Johnny. I don't think that it could happen with one instance of Johnny and multi cracking sessions since sessions will have different .johnny and different .rec files. 
So I guess that after this discussion, I can now close this task.

**Comment by math07** [comment]
> Yes, it's regarding two instances of Johnny. I don't think that it could happen with one instance of Johnny and multi cracking sessions since sessions will have different .johnny and different .rec files. 
> So I guess that after this discussion, I can now close this task.

Do you agree to close this issue ?

**Comment by shinnok** [comment]
>Do you agree to close this issue ?

Yes please close it. You should be able to do it since I made you a team member for this repo. I can also assign issues directly to you because of that.
